### PR TITLE
Main with testing code for samples functions.

### PR DESCRIPTION
**Pull request by Sharon131**
#13 #16

### CODE REVIEW

**Comment by grzanka** [comment]
@Sharon131 can you add CMake with compilation and installation instruction as well ?

**Comment by Sharon131** [comment]
> @Sharon131 can you add CMake with compilation and installation instruction as well ?

Yes, I will add that too.

File: C_samples/main.c [hunk 1]
```diff
@@ -0,0 +1,1078 @@
...
+    printf("----------------------------------------------------------------------\r\n");
+
+    printf("Proceeding with testing functions with two tables in and two tables out parameters...\r\n");
+    test_two_tables_in_two_tables_out_params_functions();
+    printf("Testing functions with two tables in and two tables out parameters ended succesfully.\r\n");
+    
+    printf("----------------------------------------------------------------------\r\n");
+
+    printf("Testing finished, all tests passed.\r\n");
+}
```
Comment (grzanka) [inline C_samples/main.c:136]: `main` is defined with `int` return type, hence it should return something

File: C_samples/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.20)
+
+project(C_samples)
+
+include_directories(${PROJECT_SOURCE_DIR})
+
+add_executable(C_samples main.c no_params_sample.c only_in_params_sample.c only_out_params_sample.c one_in_one_out_param_sample.c multiple_in_out_params_sample.c tables_params_sample.c)
```
Comment (grzanka) [inline C_samples/CMakeLists.txt:7]: consider using `GLOB`, like here:
https://github.com/libamtrack/library/blob/master/CMakeLists.txt

File: C_samples/main.c (hunk 1 above)
Comment (Sharon131) [reply C_samples/main.c:140]: Resolved.

**Comment by Sharon131** [comment]
@grzanka , how do we want to add compilation of C code to CI? Should it be a seperate workflow or just another job in the existing one? Do we want to check compilation only on Linux or maybe also on Windows?

**Comment by grzanka** [comment]
> @grzanka , how do we want to add compilation of C code to CI? Should it be a seperate workflow or just another job in the existing one? Do we want to check compilation only on Linux or maybe also on Windows?

I'd suggest another job in the existing workflow. Windows and Linux should be tested as well

**Comment by Sharon131** [comment]
Compilation and running tests of C code on Linux and Windows is added to CI.

**Review by grzanka — APPROVED** [review]
Looks OK

**Comment by grzanka** [comment]
@Sharon131 are you planning to add something or we can squash merge ?

**Pull request merged**